packages/eslint-plugin/src/rules/consistent-generic-constructors.ts

```diff
@@ -33,6 +33,30 @@ export default createRule<Options, MessageIds>({
   },
   defaultOptions: ['constructor'],
   create(context, [mode]) {
+    const isTypedArrayReference = (typeNode: TSESTree.TypeNode) => {
+      const typedArrayNames = [
+        'Int8Array',
+        'Uint8Array',
+        'Uint8ClampedArray',
+        'Int16Array',
+        'Uint16Array',
+        'Int32Array',
+        'Uint32Array',
+        'BigInt64Array',
+        'BigUint64Array',
+        'Float32Array',
+        'Float64Array',
+      ];
+      if (
+        typeNode.type === AST_NODE_TYPES.TSTypeReference &&
+        typeNode.typeName.type === AST_NODE_TYPES.Identifier &&
+        typedArrayNames.includes(typeNode.typeName.name)
+      ) {
+        return true;
+      }
+      return false;
+    };
```

Comment:
[Bug] It's not enough to just look at the name. If someone happens to redeclare their own class, such as `class Uint8Array<T>`, the rule should know to check that:

https://deploy-preview-10477--typescript-eslint.netlify.app/play/#ts=5.7.2&fileType=.ts&code=MYGwhgzhAECqCWA7ALgDgIICdNgJ4B4AVAPmgG8AoaaAehugDomKBfCikAU2WjAC44SNFhwEReAEIBXAGYzOmADLwA1p1IBeaIk4B3QSgzY8ACgCUAbnacAHgAcA9ph5kWQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0A5gMRUytTt3xwy%2BDtFQZIY6AsjgwAXxBagA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkqyARhWANqTokBMADJwMyQAugBpwNSPFgATAK4Zm7Lr06RRNKCTyZk0PADk5qAMIALdNADWZSlpno14vAENMAc3R4FHHv0hiAviD%2BQA&tokens=false

```ts
class Uint8Array<T> {
  // ...
}

let a: Uint8Array<ArrayBufferLike> = new Uint8Array();

export {}
```

You'll want to use the scope manager to see whether the identifier is referencing something declared or imported in the file vs. a global.